### PR TITLE
Add wallet JSON-RPC documentation

### LINKED ISSUE #5618: Document the wallet JSON-RPC API
Find a way to provide a OpenAPI file for the HTTPv2 API.

To investigate https://open-rpc.org/

### PR DESCRIPTION
**Pull request by ValentinTrinque**
Close #5618 

The interactive documentation can be viewed using the link:

https://playground.open-rpc.org/?schemaUrl=https://raw.githubusercontent.com/vegaprotocol/vega/9b6f9ec1be850c1cf55524b1e72878c0ffaf48ad/wallet/api/openrpc.json

### CODE REVIEW

File: wallet/api/openrpc.json [hunk 1]
```diff
@@ -0,0 +1,485 @@
...
+  "openrpc": "1.2.6",
+  "info": {
+    "version": "1.0.0",
+    "title": "Wallet API"
+  },
+  "methods": [
+    {
+      "name": "connect_wallet",
+      "summary": "Initiates a connection between a wallet and a third-party application.",
+      "description": "This method initiates a connection between a wallet and a third-party application.\n\nThe client has to review the request, and, if he accepts it, to select the wallet he wants to use for this connection.\n\nA connection token is generated and returned to the third-party application. This token is meant to be used in protected methods.\n\n**Supported connections:**\n- Multiple wallets connected for the same hostname. Each connection will have a different token.\n- A single wallet connected to multiple hostnames. Each connection will have a different token.\n- Combination of the above setups.\n\nHowever, it's not possible to have multiple connections on the same wallet for the same hostname. The previous connection will be terminated and a new token will be generated.\n\nThis method should be the entry point of every third-party applications. Once connected, see the method `get_permissions`.",
```
Comment (candida-d) [inline wallet/api/openrpc.json:11]:
```suggestion
      "description": "This method initiates a connection between a wallet and a third-party application.\n\nThe client has to review the request, and, if they accept it, select the wallet they want to use for this connection.\n\nA connection token is generated and returned to the third-party application. This token is meant to be used in protected methods.\n\n**Supported connections:**\n- Multiple wallets connected for the same hostname. Each connection will have a different token.\n- A single wallet connected to multiple hostnames. Each connection will have a different token.\n- Combination of the above setups.\n\nHowever, it's not possible to have multiple connections on the same wallet for the same hostname. The previous connection will be terminated and a new token will be generated.\n\nThis method should be the entry point of every third-party application. Once connected, see the method `get_permissions`.",
```

File: wallet/api/openrpc.json [hunk 2]
```diff
@@ -0,0 +1,485 @@
...
+      "result": {
+        "name": "permissions",
+        "schema": {
+          "$ref": "#/components/schemas/permissions"
+        }
+      },
+      "examples": [
+        {
+          "name": "Get permissions set for \"vega.xyz\"",
+          "description": "The third-party application \"vega.xyz\" wants to know the permissions that has been set on the wallet in use.",
```
Comment (candida-d) [inline wallet/api/openrpc.json:132]:
```suggestion
          "description": "The third-party application \"vega.xyz\" wants to know the permissions that have been set on the wallet in use.",
```

File: wallet/api/openrpc.json [hunk 3]
```diff
@@ -0,0 +1,485 @@
...
+        }
+      ],
+      "result": {
+        "name": "transaction_status",
+        "schema": {
+          "type": "object",
+          "properties": {
+            "receivedAt": {
+              "type": "string",
+              "description": "The date at which the API received the request to send the transaction.\n\nThe time is a quoted string in RFC 3339 format, with sub-second precision added if present.",
```
Comment (candida-d) [inline wallet/api/openrpc.json:341]:
```suggestion
              "description": "The date when the API received the request to send the transaction.\n\nThe time is a quoted string in RFC 3339 format, with sub-second precision added if present.",
```

File: wallet/api/openrpc.json [hunk 4]
```diff
@@ -0,0 +1,485 @@
...
+              "examples": [
+                "2021-02-18T21:54:42.123Z"
+              ]
+            },
+            "sentAt": {
+              "type": "string",
+              "description": "The date at which the transaction has been sent to the network.\n\nThe time is a quoted string in RFC 3339 format, with sub-second precision added if present.",
```
Comment (candida-d) [inline wallet/api/openrpc.json:348]:
```suggestion
              "description": "The date when the transaction has been sent to the network.\n\nThe time is a quoted string in RFC 3339 format, with sub-second precision added if present.",
```

File: wallet/api/openrpc.json [hunk 5]
```diff
@@ -0,0 +1,485 @@
...
+            "value": {
+              "permissions": {
+                "public_key": "read"
+              }
+            }
+          }
+        },
+        {
+          "name": "Updating permissions for \"vega.xyz\" with omitted permission",
+          "description": "The third-party application \"vega.xyz\" omits a permission during the update and the client accepts. This automatically mark the omitted permission as revoked.",
```
Comment (candida-d) [inline wallet/api/openrpc.json:401]:
```suggestion
          "description": "The third-party application \"vega.xyz\" omits a permission during the update and the client accepts. This automatically marks the omitted permission as revoked.",
```

**Review by edd — DISMISSED** [review]
Thanks for the really clear docs

**nitpick (non-blocking)**:  Some of the methods say 'The client has to review the x'. I think it's *semi* true. I think it's true enough. But can't that be disabled with an option, and couldn't this be clearer? Something like 'The user will need to approve this transaction in the wallet user interface' or something?

I left this as a comment as it's not a blocker, but a thought for later.

**Review by jeremyletang — APPROVED** [review]

**Pull request merged**